Makefile.SH

```diff
@@ -1499,7 +1499,7 @@ cscope.out cscope: $(c) $(h)
 # The README below ensures that the dependency list is never empty and
 # that when MAKEDEPEND is empty $(FIRSTMAKEFILE) doesn't need rebuilding.
 
-MAKEDEPEND = Makefile makedepend_file makedepend
+MAKEDEPEND = Makefile makedependfile makedepend
```

Comment:
I would add a comment like

> The purpose of makedependfile is to guarantee that things get properly rebuilt when makedependfile.SH is changed

Reply:
But it doesn't really explain why do this.  That hasn't been obvious to a multiple people

Reply:
Ok, I have added extra comments to help understand how the makefile works with the SH variable...

> Declare in SH variable all scripts that generate an underlying file.
> In doing so, it creates a makefile target for each underlying file.
> This target can be used as a dependency to automatically regenerate the underlying file when the script has been modified.

> Targets generating the underlying file from the scripts defined in SH variable.